frontend/pkg/frontend/context_test.go

```diff
@@ -0,0 +1,55 @@
+package frontend
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+func TestTenantIDFromContext(t *testing.T) {
+	tests := []struct {
+		name      string
+		sub       arm.Subscription
+		expected  string
+		expectErr bool
+	}{
+		{
+			name: "Valid",
+			sub: arm.Subscription{
+				Properties: &arm.Properties{
+					TenantId: stringPtr("tenant-id"),
+				},
+			},
+			expected:  "tenant-id",
+			expectErr: false,
+		},
+		{
+			name:      "Missing tenantId",
+			sub:       arm.Subscription{},
+			expectErr: true,
+		},
+	}
+
+	for _, test := range tests {
+		ctx := ContextWithSubscription(context.Background(), test.sub)
+		actual, err := TenantIDFromContext(ctx)
+		if err != nil {
+			if !test.expectErr {
+				t.Errorf("expected err to be nil, got %v", err)
+			}
+		} else {
+			if test.expectErr {
+				t.Error("expected err to be non-nil")
+			}
+
+			if actual != test.expected {
+				t.Errorf("expected %s, got %s", test.expected, actual)
+			}
+		}
+	}
```

Comment:
Minor optional suggestion: as this is a unit test, every time we find an error we could do a t.Fatalf and we can remove the elses and nesting (we also fail sooner, which is good IMHO)

Reply:
I think I get what you're saying, but I think the elses are necessary for a passing test. Like

```go
// Let's say err == test.expectErr from a test case
if err != nil {
  if !test.expectErr {}
  }
}
// If no else, it would keep checking, with the else these checks are skipped
if test.expectErr {}
if actual != test.expected {}
```

Reply:
But we can always add a "return" inside the if (just at the end, right), like we usually do in Go to remove else's ? 🙂

Reply:
I don't think `return` would work since there are multiple test cases, `continue` might in this case because I noticed I forgot to start each test case with
```go
t.Run(test.name, func(t *testing.) { ... }
```
, but if I remembered to do that `continue` wouldn't work either!

Reply:
You are right 🙂
I have opened a small PR with a refactor to use subtests and remove else: https://github.com/Azure/ARO-HCP/pull/191